redis/deprecated/redisbp/monitored_client.go

```diff
@@ -30,12 +37,47 @@ func NewMonitoredFailoverClient(name string, opt *redis.FailoverOptions) *redis.
 	return client
 }
 
+// ClusterClient extends redis cluster client with a functional Wait function
+type ClusterClient struct {
+	*redis.ClusterClient
+}
+
+// WaitArgs enclose inputs for Wait command into a struct
+type WaitArgs struct {
+	Key       string
+	NumSlaves int
+	Timeout   time.Duration
+}
+
+// Wait blocks the current client until all the previous write commands are
+// successfully transferred and acknowledged by at least the specified number of replicas.
+func (cc *ClusterClient) Wait(ctx context.Context, args WaitArgs) (replicas int64, err error) {
+	if args.NumSlaves <= 0 {
+		return 0, nil
+	}
+
+	client, err := cc.ClusterClient.MasterForKey(ctx, args.Key)
+	if err != nil {
+		return 0, fmt.Errorf("redisbp: error while trying to retrieve master from key, %w", err)
+	}
+
+	replicas, err = client.Wait(ctx, args.NumSlaves, args.Timeout).Result()
+	if err != nil {
+		return 0, fmt.Errorf("redisbp: error while trying to apply replication factor, %w", err)
+	} else if int(replicas) < args.NumSlaves {
+		return replicas, ErrReplicationFactorFailed
```

Comment:
Should this also just be `return 0, ErrReplicationFactorFailed`?

Reply:
This still returns the number of replicas it reached, it could be 0 < r < numSlaves.

Reply:
🔕 It's probably more useful to add how many replicas we got into the error message, as long as you use `%w` caller can still use `errors.Is` to check it:

```suggestion
		return replicas, fmt.Errorf("%w: %d/%d", ErrReplicationFactorFailed, replicas, args.NumReplicas)
```